src/mmw/apps/modeling/views.py

```diff
@@ -183,7 +183,9 @@ def get_job(request, job_uuid, format=None):
 
 
 def _initiate_analyze_job_chain(area_of_interest, job_id):
-    return chain(tasks.run_analyze.s(area_of_interest),
+    return chain(tasks.polygon_to_id.s(area_of_interest),
+                 tasks.id_to_histogram.s(),
+                 tasks.histogram_to_survey.s(),
                  save_job_result.s(job_id, area_of_interest)) \
         .apply_async(link_error=save_job_error.s(job_id))
```

Comment:
Yes, the `link_error` parameter is a function which is essentially a `catch` block if there are any failures. http://celery.readthedocs.org/en/latest/userguide/calling.html#linking-callbacks-errbacks